pavelib/assets.py

```diff
@@ -89,6 +90,36 @@ def applicable_sass_directories(systems=None):
     return applicable_directories
 
 
+class DebounceEventQueue(EventQueue):
+    """
+    Event queue that discards puts if they happen too quickly.
+    """
+    wait_seconds = 1
+
+    def __init__(self, *args, **kwargs):
+        EventQueue.__init__(self, *args, **kwargs)
+        self._last_updated = None
+
+    def _put(self, *args, **kwargs):
+        """
+        Only puts the item if the last put was at least `wait_seconds` ago.
+        """
+        now = datetime.now()
+        if not self._last_updated or (now - self._last_updated).total_seconds() > self.wait_seconds:
+            self._last_updated = now
+            EventQueue._put(self, *args, **kwargs)
```

Comment:
It seems like this could lead to an unfortunate circumstance of:

Time 0.0 -- Change A -> Triggers rebuild
Time 0.1 -- Change B -> Doesn't trigger

At that point, Change B won't be seen until some other change C happens.

Would it be possible to instead have a timer that gets kicked off/restarted by any change, which would wait 1 second, and after a second drop an item on the queue?

That would lead to the behavior:

Time 0.0 -- Change A -> Triggers rebuild at 1.0
Time 0.1 -- Change B -> Cancels rebuild at 1.0, triggers rebuild at 1.1.

Said another way: rather than dropping all within `wait_seconds` of a build starting, wait for `wait_seconds` after the most recent change to start a build.